codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -524,6 +524,38 @@ impl Entity {
     }
 }
 
+///  used to get the positive integers in the directives from the GraphQL schema.
+fn get_positive_integer(
+    arg_value: &Value<String>,
+    field_name: &str,
+    param_name: &str,
+) -> anyhow::Result<u32> {
+    match arg_value {
+        Value::Int(i) => {
+            let val = i.as_i64().ok_or_else(|| {
+                anyhow!(
+                    "{} value must be a positive integer. Field '{}'",
+                    param_name,
+                    field_name
+                )
+            })?;
+            if val < 0 {
+                return Err(anyhow!(
+                    "{} value must be a positive integer. Field '{}'",
+                    param_name,
+                    field_name
+                ));
+            }
+            Ok(val as u32)
+        }
+        _ => Err(anyhow!(
+            "{} value must be an integer. Field '{}'",
+            param_name,
+            field_name
+        )),
+    }
+}
```

Comment:
This function is very simple but it was getting repetitive inside the code since it was used in 3 places, and this is cleaner. 
 
 The big `from_obj_field` function should probably be broken down further too.